launch_ros/launch_ros/actions/load_composable_nodes.py

```diff
@@ -0,0 +1,202 @@
+            request.node_namespace = perform_substitutions(
+                context, composable_node_description.node_namespace
+            )
+        # request.log_level = perform_substitutions(context, node_description.log_level)
+        if composable_node_description.remappings is not None:
+            for from_, to in composable_node_description.remappings:
+                request.remap_rules.append('{}:={}'.format(
+                    perform_substitutions(context, list(from_)),
+                    perform_substitutions(context, list(to)),
+                ))
+        if composable_node_description.parameters is not None:
+            request.parameters = [
+                param.to_parameter_msg() for param in to_parameters_list(
+                    context, evaluate_parameters(
+                        context, composable_node_description.parameters
+                    )
+                )
+            ]
+        if composable_node_description.extra_arguments is not None:
+            request.extra_arguments = [
+                param.to_parameter_msg() for param in to_parameters_list(
+                    context, evaluate_parameters(
+                        context, composable_node_description.extra_arguments
+                    )
+                )
+            ]
+        response = self.__rclpy_load_node_client.call(request)
+        if not response.success:
+            _logger.error("Failed to load node '{}' in container '{}': {}".format(
+                response.full_node_name, self.__target_container.node_name, response.error_message
+            ))
+
+    def _load_in_sequence(
+        self,
+        composable_nodes_description: List[ComposableNode],
+        context: LaunchContext
+    ) -> None:
+        """
+        Load composable nodes sequentially.
+
+        :param composable_nodes_description: descriptions of composable nodes to be loaded
+        :param context: current launch context
+        """
+        next_composable_node_description = composable_nodes_description[0]
+        composable_nodes_description = composable_nodes_description[1:]
+        self._load_node(next_composable_node_description, context)
+        if len(composable_nodes_description) > 0:
+            context.add_completion_future(
+                context.asyncio_loop.run_in_executor(
+                    None, self._load_in_sequence, composable_nodes_description, context
+                )
+            )
+
+    def _load_in_parallel(
+        self,
+        composable_nodes_description: List[ComposableNode],
+        context: LaunchContext
+    ) -> None:
+        """
+        Load composable nodes in parallel.
+
+        :param composable_nodes_description: descriptions of composable nodes to be loaded
+        :param context: current launch context
+        """
+        for composable_node_description in composable_nodes_description:
+            context.add_completion_future(
+                context.asyncio_loop.run_in_executor(
```

Comment:
I think `asyncio` loops are always single threaded, so this will still load the nodes sequentially. I'm not sure about the order they'll be run, but I imagine it's the same order that `run_in_executor` is called.

Reply:
That's true as long as `_load_node()` remains synchronous, or in other words, as long as we keep using services synchronously. And of course all this is assuming that once the service response comes along, the composed node has been fully brought up already.

IMHO the neatest solution would be to turn `rclpy` service methods into coroutines. Alternatively, we could spin up a pool of threads, but I'm less fond of it. What do you think?

Reply:
> neatest solution would be to turn rclpy service methods into coroutines

Agreed that would be the neatest. I think the future returned from [`call_async`](https://github.com/ros2/rclpy/blob/28a77736f1d5ffd13a1dc065e066fab3a89f27f2/rclpy/rclpy/client.py#L113) could be `await`'d in an asyncio loop, but that future expects an rclpy executor to be spinning somewhere. I recommend removing the parallel option for now, with a plan to add it later.

Reply:
Fair enough. Want me to comment it out w/ a TODO or drop it altogether?